src/vmm/src/builder.rs

```diff
@@ -365,9 +365,12 @@ pub fn build_microvm_for_boot(
     // Move vcpus to their own threads and start their state machine in the 'Paused' state.
     vmm.start_vcpus(
         vcpus,
-        seccomp_filters
-            .remove("vcpu")
-            .ok_or_else(|| MissingSeccompFilters("vcpu".to_string()))?,
+        Arc::new(
+            seccomp_filters
+                .get("vcpu")
+                .ok_or_else(|| MissingSeccompFilters("vcpu".to_string()))?
+                .clone(),
```

Comment:
You could avoid having to `clone()` the entire _vcpu_ filter vec by changing `BpfThreadMap` definition:

```diff
- pub type BpfThreadMap = HashMap<String, BpfProgram>;
+ pub type BpfThreadMap = HashMap<String, Arc<BpfProgram>>;
```

That way any entry in the map can be easily shared among multiple users/consumers with no cloning necessary.

Reply:
Indeed 👍🏻 . I'll add a commit that does this modification

Reply:
Done, PTAL